context-free.md

```diff
@@ -2,21 +2,55 @@
 
 ##  Who is this programming language for?
 
+This programming language is for people who want to be able to quickly make
+pictures that can easily be described as a context-free grammar. Many
+recursively generated pictures fall into this category, such as different kinds
+of fractals.
+
 
 ## What is easy to do in this language? Why is it easy?
 
+It is easy to make pictures that are generated by a context-free grammar. This
+is easy because it is how the language is designed: each shape is made up of
+other shapes recursively.
+
 
 ## What is hard to do in this language? Why is it hard?
 
+It is hard(er) to make non-recursive pictures in this language. This is because
+you have to write out all the steps by hand, as opposed to letting the grammar
+do the work for you.
+
+It is also hard to write programs that are supposed to do things other than
+just making pictures, such as writing a game. The language isn't designed
+for that to really be possible, so that would be extremely hard!
+
 
 ## How did you learn how to program in this language?
 
+I looked at the examples that came with the program, as well as the
+[documentation][1].
+
+[1]: http://www.contextfreeart.org/mediawiki/index.php/CFDG_Introduction
+
 
 ## What is the underlying _computational model_ for this programming language? 
 _We don't yet have a great definition of the term "computational model". 
 For now, try to come up with the clearest, most concise explanation of what 
 happens when a ContextFree program runs._
 
+When a ContextFree program runs, it generates an element of the language whose
+grammar is encoded in the program. (A ContextFree program is essentially a
+grammar). It may make random choices about how to expand a symbol using
+weights, and while it generates the element of the language, it draws it on the
+screen.
+
 
 ## What do you think is interesting about the ContextFree program you wrote?
 
+I like color, so I found interesting how colorful my program's output was. I
+was also intrigued by what happened when I changed the rotation parameters from
+divisors of 360 (36 and 72) to non-divisors of 360 (38 and 70). Before, there
+were no spirals, but afterwards, there were! This makes sense in retrospect,
+but it was very interesting that you get better pictures with strange
+numerical parameters.
```

Comment:
Yea I found that also! If you make things neat the pictures end up less interesting. I thing CF does a really good job of making "different" interesting.